diff --git a/build_runner.dart b/build_runner.dart
index 6c4044b..4d9e08f 100644
--- a/build_runner.dart
+++ b/build_runner.dart
@@ -79,9 +79,6 @@ interface class RustBuildRunner {
       toolchainTomlPath: path.join(crateDirectory.path, 'rust-toolchain.toml'),
     );

-    logger.info('Ensuring $toolchainChannel is installed');
-    await ensureToolchainDownloaded(crateDirectory.path);
-
     logger.info('Running cargo build');
     await processRunner.invokeRustup(
       [
@@ -108,6 +105,8 @@ interface class RustBuildRunner {
       environment: {
         ...buildEnvironmentFactory.createBuildEnvVars(codeConfig),
         ...extraCargoEnvironmentVariables,
+        'CARGO_HOME': '/run/build/aria/cargo',
+        'RUSTUP_HOME': '/var/lib/rustup',
       },
     );
