diff --git a/run_builder.dart b/run_builder.dart
index 9ebfe0c..85c2324 100644
--- a/run_builder.dart
+++ b/run_builder.dart
@@ -393,8 +393,10 @@ class RunCMakeBuilder {
     }
     final defs = <String>[];
     defs.add('-DCMAKE_SYSTEM_NAME=Linux');
-    final toolchain = await linuxToolchainCmake();
-    defs.add('-DCMAKE_TOOLCHAIN_FILE=${toolchain.normalizePath().toFilePath()}');
+    if (Architecture.current != targetArchitecture) {
+      final toolchain = await linuxToolchainCmake();
+      defs.add('-DCMAKE_TOOLCHAIN_FILE=${toolchain.normalizePath().toFilePath()}');
+    }
     return defs;
   }
